### PR TITLE
Feature: delete a library

### LINKED ISSUE #20: Delete a library
## Description

blocked by #8 

Add actions column in libraries table and for each row, add a button. 
When we clicked to the button, we display the popup and on confirmation delete the library.

image
image

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #20

### CODE REVIEW

File: src/composables/Dialog.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if (event.type === 'close') {
+      show.value = false;
+      return;
+    }
+
+    show.value = true;
+
+    if (onOpen) {
+      onOpen(event);
+    }
```
Comment (k-y0u) [inline src/composables/Dialog.js:32]:
```
if (event.key === key && event.type !== 'close' && onOpen) {
  show.value = true;
  onOpen(event);
} else if (event.key === key && event.type === 'close') {
  show.value = false;
}
```

File: src/composables/Dialog.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { onMounted, onUnmounted, ref } from 'vue';
+import DialogEvent from 'src/composables/DialogEvent';
+
+/**
+ * Composable that setup specific dialog referenced by key.
```
Comment (justinehell) [inline src/composables/Dialog.js:5]: Composable that sets up specific dialog referenced by key.

File: tests/unit/composables/Dialog.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+vi.mock('src/composables/DialogEvent');
+
+describe('Test: Dialog', () => {
+  let wrapper;
+  let subscribe;
+  let unsubscribe;
+
+  /**
+   * Mount component with composable.
+   * @param {Function} callback - Callback use by the component.
```
Comment (justinehell) [inline tests/unit/composables/Dialog.test.js:15]: Callback used by the component.

File: tests/unit/composables/Dialog.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    it('should open dialog when the key is valid', () => {
+      mountComponent(() => {
+        expect(wrapper.vm.show).toBeTruthy();
+      });
+
+      DialogEvent.next({ key: 'test' });
+    });
+
+    it('should open dialog when the key is valid and event type is close', () => {
```
Comment (justinehell) [inline tests/unit/composables/Dialog.test.js:52]: should close dialog when the key is valid and event type is close

File: tests/e2e/features/Libraries.feature [hunk 5]
```diff
@@ -43,3 +56,18 @@ Feature: Test roundtrip of the application: Libraries
...
+    ################## Delete library ##################
+    ####################################################
+
+    ## 301 Should delete selected library
+    When I click on '[data-cy="library_id_2_button_remove"]'
+    Then I expect '[data-cy="button_confirm"]' exists
+
+    When I click on '[data-cy="button_confirm"]'
+    Then I expect 'positive' toast to appear with text 'Library is removed.'
+    And  I expect '[data-cy="libraries_table"] tbody tr' appear 1 times on screen
```
Comment (justinehell) [inline tests/e2e/features/Libraries.feature:70]: 1 time

**Review by justinehell — CHANGES_REQUESTED** [review]

File: tests/unit/composables/Dialog.test.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    DialogEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
+    wrapper = mount({
+      components: { useDialog },
+      setup() {
+        return useDialog('test', callback);
+      },
+      template: '<div id="dialog" v-if="show"></div>',
```
Comment (k-y0u) [inline tests/unit/composables/Dialog.test.js:30]: put it before `setup`

File: tests/unit/services/LibraryService.test.js [hunk 7]
```diff
@@ -40,11 +40,19 @@ describe('Test: LibraryService', () => {
...
-      const { api } = await import('boot/axios');
       api.get.mockImplementation(() => Promise.resolve({ data: { results: libraries } }));
 
       const data = await LibraryService.find();
       expect(data).toEqual(libraries);
     });
   });
+
+  describe('Test function: remove', () => {
+    it('should delete library', async () => {
```
Comment (k-y0u) [inline tests/unit/services/LibraryService.test.js:51]: should remove the library corresponding to the given id

File: src/components/dialog/RemoveLibraryDialog.vue [hunk 8]
```diff
@@ -0,0 +1,72 @@
+<template>
+  <q-dialog v-model="show">
+    <q-card>
+      <q-card-section class="flex row">
+        <span class="text-h6 text-center">
+          {{ $t('RemoveLibraryDialog.text.title', { name: library.name }) }}
+        </span>
+      </q-card-section>
+      <q-form @submit="onSubmit">
+        <q-card-section class="flex column flex-center">
```
Comment (k-y0u) [inline src/components/dialog/RemoveLibraryDialog.vue:10]: flex column -> column

File: src/components/tables/LibrariesTable.vue [hunk 9]
```diff
@@ -56,9 +66,22 @@ const columns = ref([{
 }]);
 const libraries = ref([]);
 
-onMounted(async () => {
+let reloadLibrariesEventRef;
+
+/**
+ * Search and display libraries.
+ */
+async function search() {
```
Comment (k-y0u) [inline src/components/tables/LibrariesTable.vue:74]: async function -> return promise

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/Dialog.js (hunk 2 above)
Comment (Zorin95670) [reply src/composables/Dialog.js:5]: done

File: tests/unit/composables/Dialog.test.js (hunk 3 above)
Comment (Zorin95670) [reply tests/unit/composables/Dialog.test.js:15]: done

File: tests/unit/composables/Dialog.test.js (hunk 4 above)
Comment (Zorin95670) [reply tests/unit/composables/Dialog.test.js:52]: done

File: tests/e2e/features/Libraries.feature (hunk 5 above)
Comment (Zorin95670) [reply tests/e2e/features/Libraries.feature:70]: done

File: tests/unit/composables/Dialog.test.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+   */
+  function mountComponent(callback = undefined) {
+    subscribe = vi.fn();
+    unsubscribe = vi.fn();
+
+    DialogEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
+    wrapper = mount({
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/composables/Dialog.test.js:25]: done

File: tests/unit/composables/Dialog.test.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      components: { useDialog },
+      setup() {
+        return useDialog('test', callback);
+      },
+      template: '<div id="dialog" v-if="show"></div>',
+    });
+    wrapper.vm.show = false;
+  }
+
+  describe('Test function onDialogEvent', () => {
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply tests/unit/composables/Dialog.test.js:35]: done

File: tests/unit/composables/Dialog.test.js (hunk 6 above)
Comment (Zorin95670) [reply tests/unit/composables/Dialog.test.js:30]: done

File: tests/unit/services/LibraryService.test.js (hunk 7 above)
Comment (Zorin95670) [reply tests/unit/services/LibraryService.test.js:51]: done

File: src/components/dialog/RemoveLibraryDialog.vue (hunk 8 above)
Comment (Zorin95670) [reply src/components/dialog/RemoveLibraryDialog.vue:10]: done

File: src/components/tables/LibrariesTable.vue (hunk 9 above)
Comment (Zorin95670) [reply src/components/tables/LibrariesTable.vue:74]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**